File: benchmarks/python/array_sum.py
def main():
    SIZE = 1000
    ITERATIONS = 10000

    # Allocate array
    arr = list(range(SIZE))

    # Sum array many times
    total = 0
    for _ in range(ITERATIONS):
        subtotal = 0
        for value in arr:
            subtotal += value
        total += subtotal

    print(f"Completed {ITERATIONS} iterations, total: {total}")

if __name__ == "__main__":
    main()

File: benchmarks/go/array_sum.go
package main

import "fmt"

func main() {
	const SIZE = 1000
	const ITERATIONS = 10000

	// Allocate array
	arr := make([]int64, SIZE)

	// Initialize array
	for i := 0; i < SIZE; i++ {
		arr[i] = int64(i)
	}

	// Sum array many times
	total := int64(0)
	for iter := 0; iter < ITERATIONS; iter++ {
		sum := int64(0)
		for i := 0; i < SIZE; i++ {
			sum += arr[i]
		}
		total += sum
	}

	fmt.Printf("Completed %d iterations, total: %d\n", ITERATIONS, total)
}
